### DISCUSSION TITLE
Update to a new version

### DISCUSSION THREAD

**Discussion by PhaNtomBek** [sinamics/ztnet#233] [Q&A] [answered]
Hello. I couldn't find any instructions for updating. Tell me how to properly update to a new version. I'm using Docker Compose. Thank you for your work.

**Comment by PhaNtomBek** [comment]
Go to installed directory and run - docker-compose pull
After run - docker-compose up -d.
If these are the right steps, then there is a problem. I had version 0.4.1. If I update to the latest version, I cannot log in. After entering the data, nothing happens. I tried all versions 0.5.x. None of them worked.
If I update 0.4.1 to 0.4.2, everything works fine.

**Comment by sinamics** [comment]
Hi
Please make sure you set the `NEXTAUTH_URL` to the canonical URL or IP of your site.
See the note highlighted
https://ztnet.network/installation/docker-compose#setup

These changes was added in v0.5.0
https://github.com/sinamics/ztnet/releases/tag/v0.5.0

**Comment by PhaNtomBek** [comment]
Hello,
I did this a long time ago when I installed version v0.4.1.
I have a domain specified. Now I checked again, everything is indicated.

![image](https://github.com/sinamics/ztnet/assets/97723224/d97c2be1-f49a-48c6-acd4-e40f604ea9dc)

![image](https://github.com/sinamics/ztnet/assets/97723224/21093a42-7b6d-4e80-83f2-c3f00fa21768)

**Reply by sinamics** [reply]
Can you try to add `NEXTAUTH_URL_INTERNAL: "http://server_ip:3000"` 
If that does not help, type `docker logs ztnet` and post them here

**Comment by PhaNtomBek** [comment]
Applying migrations to the database...
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "ztnet", schema "public" at "postgres:5432"

22 migrations found in prisma/migrations


No pending migrations to apply.
Migrations applied successfully!
Seeding the database...
Environment variables loaded from .env
Running seed command `ts-node --compiler-options {"module":"CommonJS"} prisma/seed.ts` ...
Seeding:: User Options complete!
Seeding:: Updating user ID complete!

🌱  The seed command has been executed.
Database seeded successfully!
Executing command
   ▲ Next.js 14.0.3
   - Local:        http://285b397239dd:3000
   - Network:      http://172.31.255.4:3000

 ✓ Ready in 317ms
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11730:11)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'TypeError'
  },
  url: 'http://192.168.71.11:3000/api/auth/session',
  message: 'fetch failed'
}
Socket is initializing
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11730:11)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'TypeError'
  },
  url: 'http://192.168.71.11:3000/api/auth/session',
  message: 'fetch failed'
}
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11730:11)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'TypeError'
  },
  url: 'http://192.168.71.11:3000/api/auth/session',
  message: 'fetch failed'
}
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11730:11)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'TypeError'
  },
  url: 'http://192.168.71.11:3000/api/auth/session',
  message: 'fetch failed'
}
[next-auth][error][CLIENT_FETCH_ERROR]
https://next-auth.js.org/errors#client_fetch_error fetch failed {
  error: {
    message: 'fetch failed',
    stack: 'TypeError: fetch failed\n' +
      '    at Object.fetch (node:internal/deps/undici/undici:11730:11)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'TypeError'
  },
  url: 'http://192.168.71.11:3000/api/auth/session',
  message: 'fetch failed'
}

**Comment by PhaNtomBek** [comment]
I see IP addres 192.168.71.11 in logs, but my local server address is 192.168.**1**71.11

**Reply by sinamics** [reply]
did you set the correct ip  `NEXTAUTH_URL_INTERNAL: "http://192.168.171.11:3000"` ?

**Comment by PhaNtomBek** [comment]
now NEXTAUTH_URL_INTERNAL: "http://ztnet.mydomain.com:3000"

**Comment by PhaNtomBek** [comment]
If I set a local IP address, everything starts working. But, if you click the exit button, the address in the browser changes to local. You always have to change in order to enter again.
When I published the logs, the domain was set.

**Reply by sinamics** [reply, marked as answer]
Use this setup:
```yml
NEXTAUTH_URL: "http://ztnet.mydomain.com"
NEXTAUTH_URL_INTERNAL: "http://192.168.171.11:3000"
```

**Comment by PhaNtomBek** [comment]
Great.
Now everything works as it should.
Thank you very much for the help.
